pyanaconda/modules/storage/checker/utils.py

```diff
@@ -225,6 +226,50 @@ def verify_gpt_biosboot(storage, constraints, report_error, report_warning):
                                "'biosboot' type partition."))
 
 
+def verify_opal_compatibility(storage, constraints, report_error, report_warning):
+    """ Verify the OPAL compatibility.
+
+    :param storage: a storage to check
+    :param constraints: a dictionary of constraints
+    :param report_error: a function for error reporting
+    :param report_warning: a function for warning reporting
+    """
+    if arch.get_arch() == "ppc64le" and arch.is_powernv():
+        # Check the kernel version.
+        version = _get_opal_firmware_kernel_version()
+        if version and LooseVersion(version) >= LooseVersion("5.10"):
+            return
+
+        # Is /boot on XFS?
+        dev = storage.mountpoints.get("/boot") or storage.mountpoints.get("/")
+        if dev and dev.format and dev.format.type == "xfs":
+            report_warning(_(
+                "Your firmware doesn't support XFS file system features "
+                "on the /boot file system. The system will not be bootable. "
+                "Please, upgrade the firmware or change the file system type."
+            ))
+
+
+def _get_opal_firmware_kernel_version():
+    """Get the firmware kernel version for OPAL systems.
+
+    For example: 5.10.50-openpower1-p59fd803
```

Comment:
Please, be aware that the version could also be something like "v4.15.9-openpower1-p9e03417" (see https://bugzilla.redhat.com/show_bug.cgi?id=1977236#c16) - in that case comparing of versions would throw a traceback:
```
>>> LooseVersion("5.10") > LooseVersion("v4.15.9-openpower1-p9e03417")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.9/distutils/version.py", line 64, in __gt__
    c = self._cmp(other)
  File "/usr/lib64/python3.9/distutils/version.py", line 341, in _cmp
    if self.version < other.version:
TypeError: '<' not supported between instances of 'int' and 'str'
```

Reply:
I think there are 2 options, either check the string from `ibm,firmware-versions/linux` against `<num>.<num>.<num>-something` format or wrap the comparison in a `try` block. This is what I meant with the positive identification of 5.10+ kernel. If the installer can't be sure about the version, then it's safer to expect system without new-enough XFS support.

Reply:
I have put back the code that removes `v` from the version and moved the version comparison into a try-except block.